src/datastore/datastore.js

```diff
@@ -0,0 +1,62 @@
+import { DataStore } from '@aws-amplify/datastore';
+import { Restaurant, Room } from '../models';
+
+class Database {
+    /* Creates a room with an auto-generated ID, returns the ID */
+    async createRoom(user) {
+        let id = Math.floor(Math.random() * (999999 - 100000 + 1)) + 100000;
+        await DataStore.save(new Room({
+            code: id,
+            restaurants: [],
+            users: [user],
+            current: null,
+        }));
+        return id;
+    }
```

Comment:
I know the chances are slim, but in the case that we have 2 same id's generated randomly, do we purge the room once everyone leaves?

Reply:
Good idea, I think I'll add a query that scans and removes all rooms with no users.